docs/source/notes/autograd.rst

```diff
@@ -3,119 +3,394 @@
 Autograd mechanics
 ==================
 
-This note will present an overview of how autograd works and records the
-operations. It's not strictly necessary to understand all this, but we recommend
-getting familiar with it, as it will help you write more efficient, cleaner
-programs, and can aid you in debugging.
+This note will present an overview of how autograd works. It's not strictly necessary
+to understand all this, but we recommend getting familiar with it, as it will help you
+write more efficient, cleaner programs, and can aid you in debugging.
 
-.. _excluding-subgraphs:
+TL-DR:
 
-Excluding subgraphs from backward
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+- Mathematical formulation: we compute vector Jacobian products using the chain rule. We support inplace operations using hidden versioning of the Tensors.
+- Non-gradients: we provide helper functions to be able to compute things that are not real gradients.
+- Will inplace operation work: Because they modify the original value, if anything needs this value for the backward, the backward will fail.
```

Comment:
nit: Do you mean to have a question mark? "Will inplace operation work?"

nit: "backward pass" sounds better?